console/network/src/lib.rs

```diff
@@ -144,10 +144,21 @@ pub trait Network:
     /// The maximum number of entries in a record.
     const MAX_RECORD_ENTRIES: usize = Self::MIN_RECORD_ENTRIES.saturating_add(Self::MAX_DATA_ENTRIES);
 
+    /// The maximum program size in bytes.
+    const MAX_PROGRAM_BYTE_SIZE: usize = 3_500_000; // 3.5 MB
+    /// The maximum program size in characters.
+    const MAX_PROGRAM_CHAR_SIZE: usize = 7_000_000; // 7 MB
```

Comment:
Why is this 3.5MB and 7MB?

From initial eyeballing, 1MB may seem excessive. Consider the number of deployments in a block and the implication on block size. It would be excessive for a block to be 7 * 100 == 700MB if an attacker tried to DOS storage. It would also prevent the block from being transmitted in the network protocol.

We may want to cap the expressivity at between 64KB-512KB from an initial glance. It is a worthwhile effort to formalize what this number should be empirically.

Reply:
@miazn is in the process of getting empirical data on what was actually deployed on testnet3. Afterwards I'll consider deployment costs and report back here.

Reply:
> It would be excessive for a block to be 7 * 100 == 700MB if an attacker tried to DOS storage.

Just to be on the same page, I think this is unlikely. 7MB deployments cost at least 7000 credits due to  `DEPLOYMENT_FEE_MULTIPLIER`, so we should not be seeing that many large deployments in a block already.

However, as previously discussed, lets bias towards safety for now. The string sizes of the top 20 deployed programs on testnet are at most 52KB. So I suggest to set the program limit to 100KB: 84a9a4518